### PR TITLE
feat: custom error reports.

### PR DESCRIPTION
**Pull request by alejandro0619**
#6 
Sorry for being inactive. I implemented a custom enum to wrap around the errors and then we can handle and display them easier. I also fixed the problem I previously reported on the issue. Feedback is welcomed!

### CODE REVIEW

File: src/delete/mod.rs [hunk 1]
```diff
@@ -1,29 +1,27 @@
-use anyhow::{bail, Result};
+use anyhow::Result;
 use async_trait::async_trait;
 use reqwest::{StatusCode, Url};
 
 use crate::client::FILES_ENDPOINT;
-use crate::{ErrorResponse, ImageKit};
-
+use crate::ImageKit;
+use crate::error::Error;
```
Comment (EstebanBorai) [inline src/delete/mod.rs:7]: Nitpick!

```suggestion
use crate::ImageKit;
use crate::error::Error;

```

File: src/error.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
+use thiserror;
+use reqwest::{StatusCode, header};
+use std::env::VarError;
+/// Custom error handling
+/// Check this out: https://docs.imagekit.io/api-reference/api-introduction#error-codes
+#[derive(thiserror::Error, Debug)]
```
Comment (EstebanBorai) [inline src/error.rs:6]:
```suggestion
use std::env::VarError;

/// Custom error handling
/// Check this out: https://docs.imagekit.io/api-reference/api-introduction#error-codes
#[derive(thiserror::Error, Debug)]
```

**Review by LeoBorai — COMMENTED** [review]
Looking good! Just minor nits!

File: src/error.rs
[the remark this answers is not in the document]
Comment (alejandro0619) [reply src/error.rs:1]: Done!

**Comment by LeoBorai** [comment]
@alejandro0619 closing for inactivity.

**Pull request closed without merging**